File: leetcode/268-missing-number/solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int n = (int)nums.size();
        int total = n * (n + 1) / 2;
        int sum = 0;
        for (int x : nums) sum += x;
        return total - sum;
    }
};

int main() {
    Solution sol;
    vector<int> nums = {3, 0, 1};
    cout << sol.missingNumber(nums) << endl; // 2
    return 0;
}

File: leetcode/268-missing-number/solution.py
from typing import List


class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        n = len(nums)
        total = n * (n + 1) // 2
        return total - sum(nums)


if __name__ == "__main__":
    print(Solution().missingNumber([3, 0, 1]))  # 2
